onnxruntime/core/session/allocator_adapters.cc

```diff
@@ -26,6 +27,27 @@ OrtAllocatorImplWrappingIAllocator::OrtAllocatorImplWrappingIAllocator(onnxrunti
     OrtAllocator::Reserve =
         [](OrtAllocator* this_, size_t size) { return static_cast<OrtAllocatorImplWrappingIAllocator*>(this_)->Reserve(size); };
   }
+  if (OrtAllocator::version >= kOrtAllocatorStatsMinVersion) {
+    OrtAllocator::GetStats =
+        [](const OrtAllocator* this_, OrtAllocator* allocator, char** stats) noexcept -> OrtStatusPtr {
+      API_IMPL_BEGIN
+#ifdef ORT_NO_RTTI
```

Comment:
Why can't you return stats if RTTI is disabled?

Reply:
Per @yuslepukhin comments, want to handle the situation that exceptions are disabled in no rtti case.
> When RTTI is not enabled, the process would abort. We don't want this to happen.
I suggest we return not implemented when RTTI is disbled. #Resolved

Reply:
This is because IAllocator::GetStats() throws when it is not overriden and most of the allocators do not implement the interface.

Reply:
This PR adds the throwing behaviour to GetStats by using ORT_NOT_IMPLEMENTED doesn't it? Can it simply return empty stats in the default implementation instead and the caller can ignore anything returned with num_allocs == 0? 

Side note: I hope we don't need this, but it's better to use the direct ORT_NO_EXCEPTIONS than the indirect ORT_NO_RTTI.

Reply:
Thanks! Followed the suggestion and make the following changes:
- Zero out stats in GetStats instead of raising an exception.
- Remove the checks of RTTI as they're no longer needed.
- Return empty KVP stats in the case of default implementation by checking num_allocs == 0.
- Updated API documentation.
- Updated UT.